earth2studio/models/px/dlesym.py

```diff
@@ -0,0 +1,1077 @@
+                    f"Input and output times must be of type 'timedelta64', got {times.dtype} for {name}"
+                )
+        if ocean_input_times[0] > atmos_input_times[0]:
+            raise ValueError(
+                f"Ocean input time must be equal to or before atmos input time, got {ocean_input_times[0]} and {atmos_input_times[0]}"
+            )
+        if len(atmos_input_times) < len(ocean_input_times):
+            raise ValueError(
+                f"Atmos input times must be equal to or longer than ocean input times, got {len(atmos_input_times)} and {len(ocean_input_times)}"
+            )
+        if len(atmos_output_times) < len(ocean_output_times):
+            raise ValueError(
+                f"Atmos output times must be equal to or longer than ocean output times, got {len(atmos_output_times)} and {len(ocean_output_times)}"
+            )
+
+        self.atmos_input_times = atmos_input_times
+        self.ocean_input_times = ocean_input_times
+        self.full_input_times = np.arange(
+            self.ocean_input_times[0],
+            self.atmos_input_times[-1] + 1,
+            atmos_input_times[1] - atmos_input_times[0],
+        )
+        self.atmos_output_times = atmos_output_times
+        self.ocean_output_times = ocean_output_times
+
+        # Setup the insolation and coupling times from the input and output times
+        self.atmos_sol_times = np.concatenate(
+            [self.atmos_input_times, self.atmos_output_times]
+        )
+        self.ocean_sol_times = np.concatenate(
+            [self.ocean_input_times, self.ocean_output_times]
+        )
+        n_atmos_steps = 1 + max(
+            self.atmos_model.output_time_dim // self.atmos_model.input_time_dim, 1
+        )
+        self.atmos_coupling_times = np.array(
+            [self.atmos_input_times[-1]] * n_atmos_steps
+        )
+        self.ocean_coupling_times = self.atmos_output_times
+
+    def input_coords(self) -> CoordSystem:
+        """Input coordinate system of the prognostic model
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+        return OrderedDict(
+            {
+                "batch": np.empty(0),
+                "time": np.empty(0),
+                "lead_time": self.full_input_times,
+                "variable": np.array(self.atmos_variables + self.ocean_variables),
+                "face": np.arange(12),
+                "height": np.arange(self.nside),
+                "width": np.arange(self.nside),
+            }
+        )
+
+    @batch_coords()
+    def output_coords(self, input_coords: CoordSystem) -> CoordSystem:
+        """Output coordinate system of the prognostic model
+
+        Parameters
+        ----------
+        input_coords : CoordSystem
+            Input coordinate system to transform into output_coords
+
+        Returns
+        -------
+        CoordSystem
+            Coordinate system dictionary
+        """
+
```

Comment:
output coords should also be used to validate the input_coords to make sure things look appropriate

e.g. https://github.com/NVIDIA/earth2studio/blob/main/earth2studio/models/px/fcn.py#L151